### PR TITLE
Step general-info

### PR DESCRIPTION
**Pull request by tokland**
### :pushpin: References

* **Issue:** Closes #8 

### :memo: Implementation

- Use cy.server+route+wait("@name") for more solid cypress tests.
- Install cypress-xpath (handy to select elements in React components that we cannot change)
- Use material-ui-picker for the DatePicker (mui uses the browser widgets, bad visual integration; d2-ui component has no clear button)
- Use d2-ui-forms for the FormBuilder
- Validations: 1) name, 2) dates can be both set or unset, but not just one set.

### :art: Screenshots

screenshot from 2019-02-06 11-35-19

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,16 +4,18 @@
   "version": "0.0.1",
   "license": "GPL-3.0",
   "author": "EyeSeeTea team",
-  "homepage": "./",
+  "homepage": ".",
```
[the remark this answers is not in the document]
Comment (tokland) [reply package.json]: I found the docs about this parameter and they recommended this value "." for relative paths. It works just the same. https://facebook.github.io/create-react-app/docs/deployment#serving-the-same-build-from-different-paths

File: src/components/date-picker/DatePicker.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import React from "react";
```
[the remark this answers is not in the document]
Comment (tokland) [reply src/components/date-picker/DatePicker.jsx]: This is certainly one of the generic components that will end up in d2-ui-components. Indeed, I prefer to move them by batches once its usage is stabilized.

File: src/components/org-units-selector/OrgUnitsSelector.jsx [hunk 3]
```diff
@@ -128,13 +124,13 @@ export default class OrgUnitsSelector extends React.Component {
             const newSelected = this.props.selected.concat(orgUnit.path);
             this.props.onChange(newSelected);
         }
-    }
+    };
 
-    handleChildrenLoaded(children) {
+    handleChildrenLoaded = children => {
```
[the remark this answers is not in the document]
Comment (tokland) [reply src/components/org-units-selector/OrgUnitsSelector.jsx]: It's madness. This is the de-facto official way now to write methods that you reference just by name (`onClick={this.handleChildrenLoaded}` so you don't all the `bind(this)` boilerplate. Personally, I'd prefer to use auto bind packages (ie. https://www.npmjs.com/package/auto-bind), but that's the way it is.

More info: https://codeburst.io/use-class-properties-to-clean-up-your-classes-and-react-components-93185879f688

File: README.md [hunk 4]
```diff
@@ -1,11 +1,13 @@
-## Development
-
-Setup:
+## Setup
 
 ```
 $ yarn install
 ```
 
+* Travis: set environment variable `CYPRESS_DHIS2_AUTH=admin:PASSWORD`.
```
[the remark this answers is not in the document]
Comment (tokland) [reply README.md]: Done

**Comment by adrianq** [comment]
@tokland Thanks for the changes. This is ready to be merged, however, travis has started failing. I think your changes should have no impact on travis but... I have even triggered a manual re-build in case it was a one-time error but it is still failing. Can you please have a look at it?

**Comment by tokland** [comment]
Yes, it's unrelated. Styles updated for DatePicker, TextField from d2-ui uses the old mui, DatePicker uses mui3, so styles must be matched. Mui 3 allows to style components through overrides. I'll check cypress.

**Pull request merged**